### PR TITLE
Update Quality Level 1

### PR DESCRIPTION
**Pull request by pablogs9**
Related:
- [x] https://github.com/eProsima/Micro-CDR/pull/61

TODO: 
  - [x] uncrustify 
  - [x] check Synopsis 
  - [x] CodeQL

When merged update the version to v1.2.2 and create a release.

### CODE REVIEW

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-CDR%20Manual%20Linux/34/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-CDR%20Manual%20Windows/33/)

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-CDR%20Manual%20Linux/36/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-CDR%20Manual%20Windows/35/)

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-CDR%20Manual%20Linux/37/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-CDR%20Manual%20Windows/36/)

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-CDR%20Manual%20Linux/42/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-CDR%20Manual%20Windows/37/)

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Contribution Guidelines
+
+The following documents constitutes a set of guidelines to which contributors must adhere.
```
Comment (FranFin) [inline CONTRIBUTING.md:3]:
```suggestion
The present document provides a set of guidelines to which contributors must adhere.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+Contributors must sign-off each commit by adding a `Signed-off-by: ...` line to commit messages to certify that they have the right to submit the code they are contributing to the project according to the [Developer Certificate of Origin (DCO)](https://developercertificate.org/).
+
+## Code Coverage
+
+As stated in [QUALITY.md](QUALITY.md), all contributions to the project must increase line code coverage.
+Because of this, contributors are asked to locally run a coverage assessment that ensures that code coverage has increased when compared to the latest execution of the [nightly coverage CI job](http://jenkins.eprosima.com:8080/view/Micro%20XRCE/job/Micro-CDR%20Nightly%20Master%20Linux/511/coverage/).
+
+## General considerations
+
+Any other contributing policie can be found in [eProsima general guidelines](https://github.com/eProsima/policies/blob/main/CONTRIBUTING.md).
```
Comment (FranFin) [inline CONTRIBUTING.md:36]:
```suggestion
Any other contributing policy can be found in [eProsima general guidelines](https://github.com/eProsima/policies/blob/main/CONTRIBUTING.md).
```

File: PLATFORM_SUPPORT.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+Tier 1 platforms are subjected to our unit test suite and other testing tools on a frequent basis including continuous
+integration jobs and nightly jobs.
+Errors or bugs discovered in these platforms are prioritized for correction by the development team.
+Significant errors discovered in Tier 1 platforms can impact release dates and we strive to resolve all known high
+priority errors in Tier 1 platforms prior to new version releases.
+
+## Tier 2
+
+Tier 2 platforms are subject to periodic CI testing which runs both builds and tests with publicly accessible results.
+The CI is expected to be run at least within a week of relevant changes for the current release of **Micro CDR**.
```
Comment (FranFin) [inline PLATFORM_SUPPORT.md:17]:
```suggestion
The CI is expected to be run at least within a week from the last relevant changes for the current release of **Micro CDR**.
```

File: PLATFORM_SUPPORT.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+encouraged.
+Errors may be present in released product versions for Tier 2 platforms.
+Known errors in Tier 2 platforms will be addressed subject to resource availability on a best effort basis and may or
+may not be corrected prior to new version releases.
+One or more entities should be committed to continuing support of the platform.
+
+## Tier 3
+
+Tier 3 platforms are those for which community reports indicate that the release is functional.
+The development team does not run the unit test suite or perform any other tests on platforms in Tier 3.
```
Comment (FranFin) [inline PLATFORM_SUPPORT.md:29]:
```suggestion
The development team does not run the unit test suite nor perform any other tests on platforms in Tier 3.
```

File: QUALITY.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
+This document is a declaration of software quality for **eProsima Micro CDR** based on the guidelines provided in the [ROS 2 REP-2004 document](https://www.ros.org/reps/rep-2004.html).
+
+# Quality Declaration
+
+**eProsima Micro CDR** is a C99 library that provides a serialization mechanism: the [standard CDR](https://www.omg.org/cgi-bin/doc?formal/02-06-51) serialization mechanism.
```
Comment (FranFin) [inline QUALITY.md:5]:
```suggestion
**eProsima Micro CDR** is a C99 library that functions as a serialization tool, according to the [standard CDR](https://www.omg.org/cgi-bin/doc?formal/02-06-51) serialization mechanism.
```

File: QUALITY.md [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+
+The public API is documented [in this repo README.md](https://github.com/eProsima/Micro-CDR#api-functions).
+
+### API Stability Policy [1.iv]
+
+**eProsima Micro CDR** will only break public API between major releases.
+
+### ABI Stability Policy [1.v]
+
+Any ABI break in **eProsima Micro CDR** will be done between minor versions and it should be clearly stated on the release notes.
```
Comment (FranFin) [inline QUALITY.md:32]:
```suggestion
Any ABI break in **eProsima Micro CDR** will be done between minor versions and it should be clearly stated in the release notes.
```

File: QUALITY.md [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+
+### Peer Review Policy [2.iii]
+
+All pull requests will be peer-reviewed by at least one other contributor who did not author the pull request. Approval is required before merging.
+
+### Continuous Integration [2.iv]
+
+All pull requests must pass CI to be considered for merging, unless maintainers consider that there is enough evidence that the failure is unrelated to the changes.
+CI testing is automatically triggered by incoming pull requests.
+Current nightly results can be seen here for all supported platforms:
```
Comment (FranFin) [inline QUALITY.md:60]:
```suggestion
Current nightly results for all supported platforms can be checked at the links:
```

File: QUALITY.md [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+
+### Documentation Policy [2.v]
+
+All pull requests must resolve related documentation changes before merging as stated in [CONTRIBUTING](CONTRIBUTING.md).
+
+## Documentation [3]
+
+### Feature Documentation [3.i]
+
+**eProsima Micro CDR** provides one features corresponding to its serialization mechanisms:
```
Comment (FranFin) [inline QUALITY.md:73]:
```suggestion
**eProsima Micro CDR** provides one feature corresponding to its serialization mechanisms:
```

File: QUALITY.md [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+**eProsima Micro CDR** includes a complete API Reference documented [in this repo README.md](https://github.com/eProsima/Micro-CDR#api-functions).
+
+### License [3.iii]
+
+The license for **eProsima Micro CDR** is Apache 2.0, and a summary can be found in each source file.
+A full copy of the license can be found [here](LICENSE).
+
+### Copyright Statements [3.iv]
+
+**eProsima Micro CDR** copyright holder provide a statement of copyright in each source file.
```
Comment (FranFin) [inline QUALITY.md:88]:
```suggestion
The **eProsima Micro CDR** copyright holder provides a statement of copyright in each source file.
```

File: QUALITY.md [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+
+Each part of the public API has tests, and new additions or changes to the public API require tests before being added.
+The tests aim to cover typical usage and corner cases.
+
+### Coverage [4.iii]
+
+[![Coverage](https://img.shields.io/jenkins/coverage/cobertura?jobUrl=http%3A%2F%2Fjenkins.eprosima.com%3A8080%2Fview%2FMicro%2520XRCE%2Fjob%2FMicro-CDR%2520Nightly%2520Master%2520Linux%2F)](http://jenkins.eprosima.com:8080/view/Micro%20XRCE/job/Micro-CDR%20Nightly%20Master%20Linux/)
+*eProsima Micro CDR* aims to provide a line coverage **above 90%**.
+*Micro CDR* code coverage policy comprises:
+1. All contributions to *Micro CDR* must increase (or at least keep) current line coverage.
```
Comment (FranFin) [inline QUALITY.md:111]:
```suggestion
1. All contributions to *Micro CDR* must increase (or at least keep) the current line coverage.
```

File: QUALITY.md [hunk 11]
```diff
@@ -0,0 +1,193 @@
...
+   This is done to ensure that the **90%** line coverage goal is eventually met.
+1. Line coverage regressions are only permitted if properly justified and accepted by maintainers.
+1. If the CI system reports a coverage regression after a pull request has been merged, the maintainers must study the case and decide how to proceed, mostly reverting the changes and asking for a more thorough testing of the committed changes.
```
Comment (FranFin) [inline QUALITY.md:114]:
```suggestion
1. If the CI system reports a coverage regression after a pull request has been merged, the maintainers must study the case and decide how to proceed, most often reverting the changes and asking for a more thorough testing of the committed changes.
```

File: QUALITY.md [hunk 12]
```diff
@@ -0,0 +1,193 @@
...
+2. This policy is enforced through the [nightly Micro CDR CI job](http://jenkins.eprosima.com:8080/view/Micro%20XRCE/job/Micro-CDR%20Nightly%20Master%20Linux/).
+
+As stated in [CONTRIBUTING.md](CONTRIBUTING.md), developers and contributors are asked to run a line coverage assessment locally before submitting a PR.
```
Comment (FranFin) [inline QUALITY.md:117]:
```suggestion
As stated in [CONTRIBUTING.md](CONTRIBUTING.md), developers and contributors are requested to run a line coverage assessment locally before submitting a PR.
```

File: QUALITY.md [hunk 13]
```diff
@@ -0,0 +1,193 @@
...
+
+### Performance [4.iv]
+
+**eProsima Micro CDR** does not provide performance tests.
+However, performance is indirectly tested by *eprosima Micro XRCE-DDS Client*.
+
+Any performance regression detected in *eprosima Micro XRCE-DDS Client* would be analyzed and, in case that it is related to **eProsima Micro CDR** or could be solved modifying this library, changes could be made to the library in order to revert the performance regression.
```
Comment (FranFin) [inline QUALITY.md:124]:
```suggestion
Any performance regression detected in *eProsima Micro XRCE-DDS Client* would be analyz¡sed and, in case it is related to **eProsima Micro CDR** or could be solved by modifying this library, changes can be made to the library in order to revert the performance regression.
```

File: QUALITY.md [hunk 14]
```diff
@@ -0,0 +1,193 @@
...
+
+### Linters and Static Analysis [4.v]
+
+**eProsima Micro CDR** [code style](https://github.com/eProsima/cpp-style) is enforced using [*uncrustify*](https://github.com/uncrustify/uncrustify).
+Among the CI tests there are tests that ensures that every pull request is compliant with the code style.
```
Comment (FranFin) [inline QUALITY.md:129]:
```suggestion
Among the CI tests, there are tests that ensure that every pull request is compliant with the code style.
```

File: README.md [hunk 15]
```diff
@@ -8,14 +8,14 @@
...
+*eProsima Micro CDR* is a *C* library implementing the *CDR* standard serialization methods.
 This library is focused on embedded and resource-limited systems.
 
-*MicroCDR* uses a static buffer, and allow to serialize and deserialize in both, big endianness and little endianness.
+*Micro CDR* uses a static buffer, and allow to serialize and deserialize in both, big endianness and little endianness.
 
 ## Usage examples
 This is a code example showing the serialization and deserialization of a string.
-As *MicroCDR* uses a static buffer, that means the user has to provide a defined buffer and its size during the *ucdrBuffer* creation.
+As *Micro CDR* uses a static buffer, that means the user has to provide a defined buffer and its size during the *ucdrBuffer* creation.
```
Comment (FranFin) [inline README.md:18]:
```suggestion
As *Micro CDR* uses a static buffer, the user has to provide a defined buffer and its size during the *ucdrBuffer* creation.
```

File: VERSIONING.md [hunk 16]
```diff
@@ -0,0 +1,9 @@
+Versioning policy declaration
+=============================
+
+Starting on v1.0.0, the version numbers for this library will adhere to the versioning policies defined by [Semantic Versioning](https://semver.org/).
+
+This means that API breaks should only happen between major version changes.
+If an ABI break is required, it should be done between minor versions, and it should be clearly stated on the release notes.
```
Comment (FranFin) [inline VERSIONING.md:7]:
```suggestion
If an ABI break is required, it should be done between minor versions, and it should be clearly stated in the release notes.
```

File: VERSIONING.md [hunk 17]
```diff
@@ -0,0 +1,9 @@
...
+
+The changes included on each version can be found in the [release notes](https://github.com/eProsima/Micro-CDR/releases)
```
Comment (FranFin) [inline VERSIONING.md:9]:
```suggestion
The changes included in each version can be found in the [release notes](https://github.com/eProsima/Micro-CDR/releases)
```

**Review by FranFin — CHANGES_REQUESTED** [review]

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-CDR%20Manual%20Linux/44/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-CDR%20Manual%20Windows/39/)

**Review by FranFin — APPROVED** [review]
